airbyte-integrations/connectors/source-postgres/integration_tests/seed/hook.py

```diff
@@ -0,0 +1,212 @@
+        cursor.close()
+
+def write_supporting_file(schema_name):
+    print(f"writting schema name to files: {schema_name}")
+    with open(catalog_write_file, "w") as file:
+        with open(catalog_source_file, 'r') as source_file:
+            file.write(source_file.read() % schema_name)
+    with open(catalog_view_write_file, "w") as file:
+        with open(catalog_view_source_file, 'r') as source_file:
+            file.write(source_file.read() % schema_name)
+    with open(catalog_incremental_write_file, "w") as file:
+        with open(catalog_incremental_source_file, 'r') as source_file:
+            file.write(source_file.read() % schema_name)
+    with open(abnormal_state_write_file, "w") as file:
+        with open(abnormal_state_file, 'r') as source_file:
+            file.write(source_file.read() % (schema_name, schema_name))
+    # update configs:
+    with open(secret_config_file) as base_config:
+      secret = json.load(base_config)
+      secret["schemas"] = [schema_name]
+      with open(secret_config_file, 'w') as f:
+        json.dump(secret, f)
+
+def create_table(connection, schema_name, table_name):
+    try:
+        cursor = connection.cursor()
+        # Create table
+        create_table_query = sql.SQL("""
+            CREATE TABLE IF NOT EXISTS {}.{} (
+                id VARCHAR(100) PRIMARY KEY,
+                name VARCHAR(255) NOT NULL
+            )
+        """).format(sql.Identifier(schema_name), sql.Identifier(table_name))
+
+        cursor.execute(create_table_query)
+        connection.commit()
+        print(f"Table '{schema_name}.{table_name}' created successfully")
+
+    except Exception as error:
+        print(f"Error creating table: {error}")
+        connection.rollback()
+    finally:
+        cursor.close()
+
+def generate_schema_date_with_suffix():
+    current_date = datetime.datetime.now().strftime("%Y%m%d")
+    suffix = ''.join(random.choices(string.ascii_letters + string.digits, k=8))
+    return f"{current_date}-{suffix}"
+
+def prepare():
+    schema_name = generate_schema_date_with_suffix()
+    with open("./generated_schema.txt", "w") as f:
+        f.write(schema_name)
+
+def setup():
+    schema_name = load_schema_name_from_catalog()
+    write_supporting_file(schema_name)
+    table_name = "id_and_name_hook"
+
+    # Define the records to be inserted
+    records = [
+        ('1', 'one'),
+        ('2', 'two'),
+        ('3', 'three')
+    ]
+
+    # Connect to the database
+    connection = connect_to_db()
+
+    if connection:
+        # Create the schema
+        create_schema(connection, schema_name)
+        create_table(connection, schema_name, table_name)
+
+        # Insert the records
+        insert_records(connection, schema_name, table_name, records)
+
+        # Close the connection
+        connection.close()
+
+def load_schema_name_from_catalog():
+    with open("./generated_schema.txt", "r") as f:
+        return f.read()
+
+def remove_all_write_files():
```

Comment:
do we need to perform this step? I thought when container gets destroyed after test is done, these files will be gone as they are part of the container.

Reply:
We ask the setup container to port files back to directory because config needs to be dynamically generated (to incorporate schema). thus this is needed. However this is not working right not and @clnoll is working on that!